lib/storage/data/DataWrapper.js

```diff
@@ -182,19 +182,24 @@ class DataWrapper {
             method: 'batchDelete',
         });
         const keys = [];
+        let backendName = '';
         const shouldBatchDelete = locations.every(l => {
+            // legacy sproxyd location, should fallback to using regular delete
             if (typeof l === 'string') {
                 keys.push(l);
-                return true;
+                return false;
             }
-            if (l.dataStoreName === 'sproxyd') {
-                keys.push(l.key);
+            const { dataStoreName, key } = l;
+            backendName = dataStoreName;
+            const type = this.config.getLocationConstraintType(dataStoreName);
+            if (type === 'scality') {
```

Comment:
comparing to the cloudserver changes, does this not require the `key === null` filtering?

Reply:
Ah I missed it, as I was manually porting the changes. Thanks for pointing it out.